mapper/tedge_mapper/src/mapper.rs

```diff
@@ -1,46 +1,47 @@
-use c8y_translator_lib::json::CumulocityJsonError;
-use mqtt_client::MqttClient;
+use crate::error::{ConversionError, MapperError};
+use mqtt_client::{MqttClient, Topic};
 use tokio::task::JoinHandle;
 use tracing::{debug, error, instrument};
 
-pub const IN_TOPIC: &str = "tedge/measurements";
-pub const C8Y_TOPIC_C8Y_JSON: &str = "c8y/measurement/measurements/create";
-pub const ERRORS_TOPIC: &str = "tedge/errors";
-
 #[derive(Debug)]
+pub struct MapperConfig {
+    pub in_topic: Topic,
+    pub out_topic: Topic,
+    pub errors_topic: Topic,
+}
+
 pub struct Mapper {
     client: mqtt_client::Client,
-    in_topic: mqtt_client::Topic,
-    out_topic: mqtt_client::Topic,
-    err_topic: mqtt_client::Topic,
+    config: MapperConfig,
+    converter: Box<dyn Converter<Error = ConversionError>>,
+}
+
+pub trait Converter {
+    type Error;
+
+    fn convert(&self, input: &[u8]) -> Result<Vec<u8>, Self::Error>;
```

Comment:
This convert abstraction doesn't fit well what is done for `collectd`.

* This is okay for that PR, but must be addressed by the undergoing rationalization work around the mapper and the dm-agent.

Reply:
:cry: